tests/CustomMatchers/CustomMatchersForGrouping.ts

```diff
@@ -0,0 +1,60 @@
+import type { Field } from '../../src/Query/Filter/Field';
+
+declare global {
+    namespace jest {
+        interface Matchers<R> {
+            toSupportGroupingWithProperty(property: string): R;
+        }
+
+        interface Expect {
+            toSupportGroupingWithProperty(property: string): any;
+        }
+
+        interface InverseAsymmetricMatchers {
+            toSupportGroupingWithProperty(property: string): any;
+        }
+    }
+}
+
+export function toSupportGroupingWithProperty(field: Field, property: string) {
+    const fieldGrouper = field.createGrouper();
+
+    const groupingIsSupported = field.supportsGrouping();
+    const propertyAsExpected = fieldGrouper.property === property;
+
+    if (groupingIsSupported && propertyAsExpected) {
+        return {
+            message: () =>
+                `'${field.fieldName()}' field supports grouping with property set to '${fieldGrouper.property}'.`,
+            pass: true,
+        };
+    }
+
+    if (groupingIsSupported && !propertyAsExpected) {
+        return {
+            message: () =>
+                `'${field.fieldName()}' field supports grouping but property set to '${property}', expected '${
+                    fieldGrouper.property
+                }'.`,
+            pass: false,
+        };
+    }
+
+    if (!groupingIsSupported && propertyAsExpected) {
+        return {
+            message: () =>
+                `'${field.fieldName()}' field doesn't support grouping, while property is set to '${
+                    fieldGrouper.property
+                }' as expected.`,
+            pass: false,
+        };
+    }
+
+    return {
+        message: () =>
+            `'${field.fieldName()}' field doesn't support grouping and property set to '${property}', expected '${
+                fieldGrouper.property
+            }'.`,
+        pass: false,
+    };
```

Comment:
### Summary:

Please could this be changed to only report things that are **incorrect**, and not things that are correct. See heading below for more info.

Also, could you break the code intentionally and check that the output is the correct way round.

For example, when I made this change:

```diff
Index: src/Query/Filter/Field.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Query/Filter/Field.ts b/src/Query/Filter/Field.ts
--- a/src/Query/Filter/Field.ts	(revision 6610822a22946a054f0e6674f687f5e45f52558c)
+++ b/src/Query/Filter/Field.ts	(date 1682154936499)
@@ -281,6 +281,6 @@
      * Later, this will probably be moved to the {@link Field} classes.
      */
     public createGrouper(): Grouper {
-        return new Grouper(this.fieldNameSingular(), this.grouper());
+        return new Grouper('wibble', this.grouper());
     }
 }
```

The output was:

```text
Error: 'due' field supports grouping but property set to 'due', expected 'wibble'.
```

So it's reporting the expected and actual the wrong way round.

### Detail

This is valiant effort, from the very rushed minimal overview of Jest custom matchers that I gave you when we last paired on the Tasks code.

To test the outputs, I intentionally broke the code and found that the messages I got were a bit confusing - and I was trying to work out why. For example:

```text
Error: 'due' field supports grouping but property set to 'due', expected 'wibble'.
```

On reflection, it is that the output of tests should be things that are correct and need acting upon.

This starts by saying `Error: 'due' field supports grouping` - so my first reaction is, "oh, OK, I had better make it not support grouping then".

It would be more immediately understandable if it said something like:

```text
Error: property set to '...actual value...', expected '...expected value...'.
```

Reply:
Thanks a lot for your detailed comment. I'm sorry to waste lots of your time on this minor matter, hope you see it as an investment that will pay out in future =)

I modified the matcher as you proposed, if I got it correctly. I tried the error cases when I was writing the first version of the matcher, but I must have gotten confused in the 2 variables.

The PR is now ready for the review, please resolve this conversation if this is ok now.